src/lib.rs

```diff
@@ -113,51 +113,85 @@ where
+        while let Some(current_id) = self.next_id {
+            if current_id.is_last() {
                 self.next_id.take();
+                if recoverable_errors > 0 {
+                    log::warn!(
+                        "Iteration completed with {} recoverable errors.",
+                        recoverable_errors
+                    );
+                }
                 return None;
             }
-        };
 
-        if next_record_id == current_id {
-            log::error!("Got duplicate SDR record IDs! Stopping iteration.");
-            self.next_id.take();
-            return None;
-        } else {
-            self.next_id = Some(next_record_id);
+            let next_record = self
+                .ipmi
+                .send_recv(sdr::GetDeviceSdr::new(None, current_id));
+
+            match next_record {
+                Ok(record) => {
+                    if recoverable_errors > 0 {
+                        log::warn!(
+                            "Skipped {} recoverable errors before processing record 0x{:04X}.",
+                            recoverable_errors,
+                            current_id.value()
+                        );
+                    }
+
+                    let next_record_id = record.next_entry;
+
+                    if next_record_id == current_id {
+                        log::error!("Got duplicate SDR record IDs! Stopping iteration.");
+                        self.next_id.take();
+                        return None;
+                    }
+
+                    self.next_id = Some(next_record_id);
+                    return Some(record.record);
+                }
+                Err(IpmiError::ParsingFailed {
+                    error: ParseResponseError::Parse((e, next_record_id)),
+                    ..
+                }) => {
+                    recoverable_errors += 1;
+                    if recoverable_errors > MAX_RECOVERABLE_ERRORS {
+                        log::error!(
+                            "Exceeded maximum recoverable errors ({}). Stopping iteration.",
+                            MAX_RECOVERABLE_ERRORS
+                        );
+                        self.next_id.take();
+                        return None;
+                    }
+
+                    log::debug!(
+                        "Recoverable error while parsing SDR record 0x{:04X}: {e:?}. Skipping to next.",
+                        current_id.value()
+                    );
+                    self.next_id = Some(next_record_id);
+                    continue; // skip the current one
+                }
+                Err(e) => {
+                    log::error!(
+                        "Unrecoverable error while parsing SDR record 0x{:04X}: {e:?}",
+                        current_id.value()
+                    );
+                    self.next_id.take();
+                    return None;
+                }
+            }
+        }
+
+        if recoverable_errors > 0 {
```

Comment:
I don't think this message is necessary, since it'll only be printed in very specific cases and every time you would call `next()` on the iterator if it is already finished.

Please remove it, or (lightly) motivate why we should keep it.